cishouseholds/pipeline/high_level_transformations.py

```diff
@@ -2553,7 +2566,88 @@ def add_pattern_matching_flags(df: DataFrame) -> DataFrame:
         column_name_to_assign="is_self_employed",
         debug_mode=False,
     )
+    df = assign_regex_from_map(
+        df=df,
+        column_name_to_assign="regex_derived_job_sector",
+        reference_columns=["work_main_job_title", "work_main_job_role"],
+        roles=roles_map,
+    )
+    df = df.withColumn("healthcare_area", F.lit(None))
+    for healthcare_type, roles in healthcare_classification.items():  # type: ignore
+        df = df.withColumn(
+            "healthcare_area",
+            F.when(array_contains_any("regex_derived_job_sector", roles), healthcare_type).otherwise(  # type: ignore
+                F.col("healthcare_area")
+            ),
+        )
+
+    # TODO: need to exclude healthcare types from social care matching
+    df = df.withColumn("social_care_area", F.lit(None))
+    for social_care_type, roles in social_care_classification.items():  # type: ignore
+        df = df.withColumn(
+            "social_care_area",
+            F.when(array_contains_any("regex_derived_job_sector", roles), social_care_type).otherwise(  # type: ignore
+                F.col("social_care_area")
+            ),
+        )
+
+    # add boolean flags for working in healthcare or socialcare
+    df = df.withColumn("works_healthcare", F.col("healthcare_area").isNotNull())
+    df = df.withColumn("works_social_care", F.col("social_care_area").isNotNull())
+
+    df = assign_regex_match_result(
+        df=df,
+        columns_to_check_in=["work_main_job_title", "work_main_job_role"],
+        column_name_to_assign="is_patient_facing",
+        positive_regex_pattern=patient_facing_pattern.positive_regex_pattern,
+        negative_regex_pattern=patient_facing_pattern.negative_regex_pattern,
+    )
 
+    # df = df.withColumn(
+    #     "works_healthcare",
+    #     (array_contains_any("regex_derived_job_sector", healthcare_positive_roles))
+    #     & (~array_contains_any("regex_derived_job_sector", healthcare_negative_roles)),
+    # )
+
+    df = df.withColumn(
+        "is_patient_facing",
+        F.when(
+            (F.col("works_healthcare") | F.col("is_patient_facing"))
+            & (~array_contains_any("regex_derived_job_sector", patient_facing_classification["N"])),
+            True,
+        ).otherwise(False),
+    )
+
+    df = assign_column_value_from_multiple_column_map(
+        df,
+        "work_health_care_patient_facing",
+        [
+            ["No", [False, None]],
+            ["Yes, primary care, patient-facing", [True, "Primary"]],
+            ["Yes, secondary care, patient-facing", [False, "Primary"]],
```

Comment:
all patient-facing strings should have 'True' for is_patient_facing, and healthcare_area should align with string.